src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -446,7 +447,8 @@ void GcsServer::InitKVManager() {
   if (storage_type_ == "redis") {
     instance = std::make_unique<RedisInternalKV>(GetRedisClientOptions());
   } else if (storage_type_ == "memory") {
-    instance = std::make_unique<MemoryInternalKV>(main_service_);
+    instance = std::make_unique<StoreClientInternalKV>(
+        std::make_unique<InMemoryStoreClient>(main_service_));
   }
```

Comment:
Why not just use store clients here? Also for redis.

Reply:
yeah I think this will mostly impact Ant. cc @raulchen to see if we should turn it on by default.